### ISSUE TITLE
JUnite reporter xml results are not within JUnit reporting standards

### ISSUE THREAD

**Issue by wbruce31** [postmanlabs/newman#1426]
1. Newman Version (can be found via `newman -v`): `3.8.3`
 2. OS details (type, version, and architecture): using Docker image: `postman/newman_ubuntu1404`
 3. Are you using Newman as a library, or via the CLI? CLI in Docker `postman/newman_ubuntu1404`
 3. Did you encounter this recently, or has this bug always been there: I'm new to the tool
 4. Expected behaviour: JUnit reporter creates properly formatted XML file 
 5. Command / script used to run Newman: `newman run "https://api..." --environment="https://api..." -x --reporters junit --reporter-junit-export ./test-reports/bunker-api-test-results.xml`
 6. Sample collection, and auxiliary files (minus the sensitive details): super simple collection with one endpoint and 3 tests. I have one test intentionally failing.
 7. Screenshots (if applicable): Below is the resulting XML file generated.
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="bunker-service-api" time="0.963">
  <testsuite name="BusinessEntityType / BusinessEntityType" id="6c63aa1e-1fb2-4146-b555-0ecdf25d0d15" tests="3" time="0.963">
    <testcase name="BusinessEntityType has required fields" time="0.963"/>
    <testcase name="6 entity types are returned" time="0.963"/>
    <testcase name="the tests are now failing" time="0.963">
      <failure type="AssertionFailure">
        <![CDATA[Failed 1 times.]]>
      </failure>
    </testcase>
  </testsuite>
</testsuites>
```
The above XML file is what is generated. I am working with Bitbucket Pipelines which should pick-up the JUnit xml file and report on it. However, pipelines out says:
```
Located test suite containing 3 tests, with 0 failures and 0 errors.
Finished scanning for test reports. Found 1 test report files.
Merged test suites, total number tests is 3, with 0 failures and 0 errors.
Uploading test results
Finished uploading test results.
```
Bitbucket support says that the XML generated should look more like the following in order to be compliant with http://help.catchsoftware.com/display/ET/JUnit+Format
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="bunker-service-api" time="0.963">
  <testsuite name="BusinessEntityType / BusinessEntityType" id="6c63aa1e-1fb2-4146-b555-0ecdf25d0d15" tests="3" failures="1" errors="0" time="0.963">
    <testcase name="BusinessEntityType has required fields" time="0.963"/>
    <testcase name="6 entity types are returned" time="0.963"/>
    <testcase classname="JUnitXmlReporter.constructor" name="the tests are now failing" time="0.963">
      <failure type="AssertionFailure">
        <![CDATA[Failed 1 times.]]>
      </failure>
    </testcase>
   </testsuite>
</testsuites>
```
Basically, the testsuite and testcase should look more like this.
`<testsuite name="BusinessEntityType / BusinessEntityType" id="6c63aa1e-1fb2-4146-b555-0ecdf25d0d15" tests="3" failures="1" errors="0" time="0.963">`
`<testcase classname="JUnitXmlReporter.constructor" name="the tests are now failing" time="0.963">`

**Comment by kunagpal** [comment]
@wbruce31 This has been fixed with Newman v3.9.4 😄

**Comment by Al54321** [comment]
Hi,

Using Newman 3.9.4, the classname value seems to always be set to `JUnitXmlReporter.constructor`, which I'm not sure is suitable. Please could you give the following your consideration?

Based on the example from http://help.catchsoftware.com/display/ET/JUnit+Format above, the required pattern seems to be that the classname property of a testcase should be set to the testcase's parent testsuite's name. I have used caps lock to emphasise here using their example:
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
   <testsuite name="JUnitXmlReporter" errors="0" tests="0" failures="0" time="0" timestamp="2013-05-24T10:23:58" />
   <testsuite name="JUNITXMLREPORTER.CONSTRUCTOR" errors="0" skipped="1" tests="3" failures="1" time="0.006" timestamp="2013-05-24T10:23:58">
      <properties>
         <property name="java.vendor" value="Sun Microsystems Inc." />
         <property name="compiler.debug" value="on" />
         <property name="project.jdk.classpath" value="jdk.classpath.1.6" />
      </properties>
      <testcase classname="JUNITXMLREPORTER.CONSTRUCTOR" name="should default path to an empty string" time="0.006">
         <failure message="test failure">Assertion failed</failure>
      </testcase>
      <testcase classname="JUNITXMLREPORTER.CONSTRUCTOR" name="should default consolidate to true" time="0">
         <skipped />
      </testcase>
      <testcase classname="JUNITXMLREPORTER.CONSTRUCTOR" name="should default useDotNotation to true" time="0" />
   </testsuite>
</testsuites>
```

Focusing on the practical implications, the classname value seems to affect how failing test cases are grouped in reporting tools. I'm using Jenkins' JUnit reporter and, using my Newman JUnit results with classname always being set to `JUnitXmlReporter.constructor`, the failing tests get grouped under this separate category rather than being presented in relation to their containing suite, which doesn't make very much sense, and suggests that the xml information isn't quite right.

Thank you for reading, and please let me know if there's any information or clarification that would be useful.

**Comment by Al54321** [comment]
Hi @kunagpal, please could this issue be re-opened based on the above comment? The classname always being set to the same value results in very poorly structured reporting of failing tests, so the JUnit reporting standard isn't being followed correctly. Thanks!

**Comment by kunagpal** [comment]
@Al54321 This is being fixed in https://github.com/postmanlabs/newman/pull/1612

**Comment by Al54321** [comment]
@kunagpal That's brilliant, thanks for turning it around so quickly too!

**Comment by atqmetry** [comment]
Not sure whether I should create a separate issue for this, but only failed test cases have _classname_ in XML. I've just verified this with newman _4.2.2_ on Windows 10 64-Bit.

AFAIK JUnit XML format has classname in all test cases irrespective of their status. We are having trouble with such implementation. _Is it possible to add classname attribute for all test case elements in nearest future or this is intentionally created so?_ 

Please correct me if I'm missing anything here. Looking forward to your response.
